raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -101,6 +101,10 @@ protected LeaderState(
      * @return the remainingMs before the checkQuorumTimer expired
      */
     public long timeUntilCheckQuorumExpires(long currentTimeMs) {
+        // if there's only 1 voter, it should never get expired.
+        if (voterStates.size() == 1) {
+            return Long.MAX_VALUE;
+        }
         checkQuorumTimer.update(currentTimeMs);
```

Comment:
I was thinking we don't need `checkQuorumTimer` for 1 voter's case, but since in next release, we're going to allow dynamically scaling up/down voters, we should keep the flexibility here.

Reply:
Make sense @showuon . What's the side effect without this change? The quorum becomes unavailable for the election timeout?

Reply:
The behavior without this fix is described in https://issues.apache.org/jira/browse/KAFKA-16163

Reply:
@jsancio , the impact is that when "there is only 1 controller node, or only 1 combine node", the controller will keep resigning its leadership and then elect itself on every `fetch timeout`.